### ISSUE TITLE
Context menu submenu is not openable when parented to another user

### ISSUE THREAD

**Issue by Daniel-Luke-Taylor** [Yellow-Dog-Man/Resonite-Issues#1822]
### Describe the bug?

When a context menu submenu component is pressed and aimed at a submenu located under another user it does not open. The same submenu will open if the submenu root is not parented to another user. 
This becomes problematic when the submenu is located on a grabbable object and the object is picked up frequently so the submenu appears to not work sometimes to the user. 

### To Reproduce

Create a context menu submenu button
Point the submenu root at a slot located underneath another user that contains context menu buttons
The menu will not open

### Expected behavior

The submenu will open regardless if it parented under a user. 

### Resonite Version Number

2024.4.22.1366

### What Platforms does this occur on?

Windows, Linux

### Log Files

N/A

### Additional Context

Optionally addition to make the radial menu components more inline with other buttons an option in the components similar to other button components that can filter usage to "only active user", "active user when available", "exclude active user", etc. 

### Reporters

Axius

**Comment by Daniel-Luke-Taylor** [comment]
> > aimed at a submenu located under another user
> 
> Can you please explain what you're trying to build/what your underlying issue is? This sounds like a weird setup.
> 
> A user X`is parented to another user`Y`, and `Y`is trying to open a submenu that is on`X`? Am I understanding that correctly?

It's a simple setup, the submenu button opens a submenu root on another object for remote control. A user isn't parented to another user. It's simply a set a context menu buttons for a submenu on an object that is grabbable. Say for example there is a TV remote that opens a submenu that is generated on the tv, and the TV is picked up by another user. The remote will no longer be able to open/access the submenu on the tv object. 
It is an unorthodox setup compared to the standard of having the the submenu button on the same object as the submenu root. But useful in such situations where there are multiple "remotes".

**Comment by Daniel-Luke-Taylor** [comment]
The holder of the object is able to use the submenu yes. 
If it is by design then it would be nice if that design was more inline with other buttons components for example the suggestion in additional context. Regular button components have active user filters to disallow use by other users, you can press other button components that other users are holding and can filter this interaction with the active user filter if required.
Your suggestion to take the submenu root out of the grabbable object would work for the example, but would separate the menu buttons from the object they are controlling making it messy to ensure references are not broken, etc.

**Comment by Frooxius** [comment]
Yes, this is intentional behavior, if it's under another user, it's excluded.

This is done for a few reasons - one of them so you don't see context menu options of other users that might be in your hierarchy - e.g. when they're in user anchors.

We could look to refine this behavior so it allows for some scenarios, but we'd need to know more about the setup you have.

One question here is - why do you want the context menu to open for another user, when there's a different user currently holding the item?

How is the context menu in this case even being invoked for the item?

**Comment by Daniel-Luke-Taylor** [comment]
> Yes, this is intentional behavior, if it's under another user, it's excluded.
> 
> This is done for a few reasons - one of them so you don't see context menu options of other users that might be in your hierarchy - e.g. when they're in user anchors.
> 
> We could look to refine this behavior so it allows for some scenarios, but we'd need to know more about the setup you have.
> 
> One question here is - why do you want the context menu to open for another user, when there's a different user currently holding the item?
> 
> How is the context menu in this case even being invoked for the item?


Alright. In my case I have a touch button on my avatar that I want to open a context menu for options using a `ContextMenuSubmenu` component. 
The touch button with `ContextMenuSubmenu` works if I click it myself but if others click it they cannot see the submenu because all the `ContextMenuItemSource`s are on my avatar, parented to me. 
For clarification this is NOT using a `RootContextMenuItem` component. Only `ContextMenuSubmenu` and `ContextMenuItemSource`s. 
I wish to the emulate the design sensibilities that are established with other features that work like this i.e `AvatarAnchorTouchTrigger` to get the anchor/cancel submenu or clicking an empty avatar to equip/cancel. 
E.g you can click an `AvatarAnchorTouchTrigger` parented to another user and this works as expected, pulling up the submenu. 
I cannot make this work for custom context buttons because they do not show up for others if the `ContextMenuItemSource`s are parented to me. 

My proposed solution would be to add the `ActiveUserFilter` options like other button components have to the `ContextMenuItemSource`s so you can set them to only show up/generate the buttons for active user/active user when available/exclude active user/disabled/etc. 
If the default is set to active user when available then the `ContextMenuItemSource`s will still act as they do now, and the option to disable this filter will be useful to copy the same uix design that the other common interactions equip avatar and `AvatarAnchorTouchTrigger`  have for user defined options/features on avatars or other such user parented objects/etc.

**Comment by Daniel-Luke-Taylor** [comment]
> We have a "close context menu" node, but it sounds like it could be useful to have an "open context menu" node which points to a ContextMenuItem and would effectively forward/proxy clicking that object to that context menu item.

While a flux node to open a context menu would certainly be useful, the `ContextMenuSubmenu` works on other button components as well as context menu buttons. In this case I am using a touch button component, and the menu opens regularly, just not in the situation where the `ContextMenuItemSource`s the button click invokes are parented to another user. 

The issue I am having is that in order to successfully have it open the submenu the `ContextMenuItemSource`s cannot be parented to another user. 

This means while I am able to open the menu on the touch button click in order for other users to open the menu it would have to be removed from my avatar at the moment. 
The current work around is having the `ContextMenuItemSource`s reparent to root on button hover, before a user clicks the touch button. This, obviously, is not ideal as it requires me to manage the slot containing the `ContextMenuItemSource`s and because the slot is moved from my avatar I can no longer use any dynamic spaces.

**Comment by eth0fox** [comment]
FWIW: I ran up into this issue as I wanted to have a clickable thing in my nametag which would open a context menu (instead of cluttering up the name tag itsself with multiple buttons). Not being able to allow other users to open context menus on other people's avatars prevented me from doing this.

**Comment by Daniel-Luke-Taylor** [comment]
> FWIW: I ran up into this issue as I wanted to have a clickable thing in my nametag which would open a context menu (instead of cluttering up the name tag itsself with multiple buttons). Not being able to allow other users to open context menus on other people's avatars prevented me from doing this.

So while I would still like this solved properly, maybe with a tick box to toggle users being able to open them, I have found a work around for now.

The button click to open the context menu also puts the context menu in the root slot, and this happens fast enough that the sub menu can be opened.
The sub menu buttons then all parent it back to the original slot, with an additional timeout to make sure it goes back if the user exits the context menu without clicking, or you can reparent on fire on false context menu open.

**Comment by TisFoolish** [comment]
> We could look to refine this behavior so it allows for some scenarios, but we'd need to know more about the setup you have.

Personally I'd love for the ability to let anyone change my context menu options by picking me up and opening their context menu. Sounds like fun. 

Definitely shouldn't be default, though.

**Comment by Dusty-Sprinkles** [comment]
I upvoted this issue but it is definitely a major pain point when wanting to expose settings on my avatar for other users to see/play with I have to basically completely reinvent the wheel of what I already did. 

I think it would be really nice to just have a general "Allow this item to be shown" component or toggle for context menus so that they can be found by the submenu component when it includes/targets them.
